### PR TITLE
Cloud 3242 ref docs refactor

### PR DESCRIPTION
**Pull request by pedanticdev**

### CODE REVIEW

File: docs/modules/reference/pages/manage/app/detail.adoc [hunk 1]
```diff
@@ -1,124 +0,0 @@
-= Navigating the Application Detail Page
```
Comment (rohan267) [inline docs/modules/reference/pages/manage/app/detail.adoc:1]: This move broke the navigation link mentioned in     "Application Management.adoc"

File: docs/modules/reference/pages/billing/subscription/user-subscription.adoc [hunk 2]
```diff
@@ -1,50 +0,0 @@
-= Manage Users
```
Comment (rohan267) [inline docs/modules/reference/pages/billing/subscription/user-subscription.adoc:1]: This move broke the link mentioned in "Billing Management.adoc" in How-to-Guides

**Review by rohan267 — CHANGES_REQUESTED** [review]

File: docs/modules/reference/pages/billing/subscription/user-subscription.adoc (hunk 2 above)
Comment (pedanticdev) [reply docs/modules/reference/pages/billing/subscription/user-subscription.adoc:1]: I pushed a big change. I'll redo the navigation tomorrow and you take another look. Stepping out to get some medications.

Comment (rohan267) [reply docs/modules/reference/pages/billing/subscription/user-subscription.adoc]: CleanShot 2024-09-10 at 15 42 17
Not able to see "How To Guides" and "Getting Started" columns

**Comment by rohan267** [comment]
This suggestion is not related to current change, but good to include it within this changeset. It is better to have 1-2 liner definition of `application-management` in Reference doc, just like it has for `subscription and billing management`

CleanShot 2024-09-13 at 11 37 32
CleanShot 2024-09-13 at 11 38 36

**Review by rohan267 — CHANGES_REQUESTED** [review]

**Comment by pedanticdev** [comment]
Ping @rohan267

**Review by rohan267 — APPROVED** [review]

**Pull request merged**